common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/SmartcardCertPickerDialog.java

```diff
@@ -0,0 +1,222 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui.webview.challengehandlers;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+import android.widget.RadioButton;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+
+import com.microsoft.identity.common.R;
+
+import java.util.List;
+
+/**
+ * Builds and shows a dialog that allows the user to select a certificate they would like to use to authenticate.
+ */
+public class SmartcardCertPickerDialog extends SmartcardDialog {
+
+    private final List<ClientCertAuthChallengeHandler.YubiKitCertDetails> mCertList;
+    private PositiveButtonListener mPositiveButtonListener;
+    private NegativeButtonListener mNegativeButtonListener;
+    private CancelCbaCallback mCancelCbaCallback;
+
+    /**
+     * Creates new instance of SmartcardCertPickerDialog.
+     * @param certList List of ClientCertAuthChallengeHandler.YubiKitCertDetails compiled from certificates on YubiKey.
+     * @param activity Host activity.
+     */
+    public SmartcardCertPickerDialog(List<ClientCertAuthChallengeHandler.YubiKitCertDetails> certList, Activity activity) {
+        super(activity);
+        mCertList = certList;
+        mPositiveButtonListener = null;
+        mNegativeButtonListener = null;
+        mCancelCbaCallback = null;
+        createDialog();
+    }
+
+    /**
+     * Builds an AlertDialog that displays the details of the certificates in a single choice ListView and prompts the user to choose a certificate to proceed.
+     */
+    protected void createDialog() {
```

Comment:
This is create and Show dialog? Or only create?

Reply:
Only create. `show()` is inherited from the `SmartcardDialog` class in `SmartcardCertPickerDialog`